akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala

```diff
@@ -864,6 +864,16 @@ final case class Server(products: immutable.Seq[ProductVersion]) extends jm.head
 // https://tools.ietf.org/html/rfc6797
 object `Strict-Transport-Security` extends ModeledCompanion[`Strict-Transport-Security`] {
   def apply(maxAge: Long, includeSubDomains: Option[Boolean]) = new `Strict-Transport-Security`(maxAge, includeSubDomains.getOrElse(false))
+
+  def fromDirectives(directives: StrictTransportSecurityDirective*) = {
+    val maxAgeDirectives = directives.filter(_.isInstanceOf[MaxAge])
+    require(maxAgeDirectives.size == 1, "exactly one 'max-age' directive required")
+
+    val includeSubDomainsDirectives = directives.filter(_.equals(IncludeSubDomains))
```

Comment:
Since we don't use anything but the count of instances `.count(_ eq IncludeSubdomains)` would avoid creating a collection instance, make sure to use `eq` since instance equality is enough, and same here about caching the lambda.